src/main/java/com/github/webicitybrowser/spec/fetch/imp/FetchEngineImp.java

```diff
@@ -41,8 +45,36 @@ private FetchResponse httpNetworkFetch(FetchParams params) {
 
 	private void fetchResponseHandover(FetchParams params, FetchResponse response) {
 		if (params.consumeBodyAction() != null) {
-			// TODO: Queue a response
+			if(response.body() == null) {
+				queueAFetchTask(() -> params.consumeBodyAction().execute(response, true, new byte[] {}), params);
+			} else {
+				fullyReadBody(params, response);
+			}
 		}
 	}
-	
+
+	private void queueAFetchTask(Runnable fetchTask, FetchParams params) {
+		params.taskDestination().enqueue(fetchTask);
+	}
+
+	private void fullyReadBody(FetchParams params, FetchResponse response) {
+		//TODO: implement fully read
+		params.taskDestination().dequeueAll();
```

Comment:
The task destination does not get cleared, so this line should be removed
As well as the TODO line above

Reply:
This is weird because there is in specs that if task destination is null set taskDestination to be result of starting new parallel queue, but this queue is inside taskDestination. And starting new parallel queue also deques from parallelQueue's queue